posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -860,16 +895,38 @@ async def run(self, inputs: RunWorkflowInputs) -> Results:
             ),
         )
 
+        # Run the DAG
         run_model_activity_inputs = RunDagActivityInputs(team_id=inputs.team_id, dag=dag)
-        results = await temporalio.workflow.execute_activity(
-            run_dag_activity,
-            run_model_activity_inputs,
-            start_to_close_timeout=dt.timedelta(hours=1),
-            heartbeat_timeout=dt.timedelta(minutes=1),
-            retry_policy=temporalio.common.RetryPolicy(
-                maximum_attempts=1,
-            ),
-        )
+        try:
+            results = await temporalio.workflow.execute_activity(
+                run_dag_activity,
+                run_model_activity_inputs,
+                start_to_close_timeout=dt.timedelta(hours=1),
+                heartbeat_timeout=dt.timedelta(minutes=1),
+                retry_policy=temporalio.common.RetryPolicy(
+                    maximum_attempts=1,
+                ),
+                cancellation_type=temporalio.workflow.ActivityCancellationType.TRY_CANCEL,
+            )
+        except temporalio.exceptions.ActivityError as e:
+            workflow_id = temporalio.workflow.info().workflow_id
+            workflow_run_id = temporalio.workflow.info().run_id
+            try:
+                await temporalio.workflow.execute_activity(
+                    cancel_jobs_activity,
+                    CancelJobsActivityInputs(
+                        workflow_id=workflow_id, workflow_run_id=workflow_run_id, team_id=inputs.team_id
+                    ),
+                    start_to_close_timeout=dt.timedelta(minutes=5),
+                    retry_policy=temporalio.common.RetryPolicy(
+                        maximum_attempts=1,
+                    ),
```

Comment:
Any reason not to retry forever? This is just talking to our database, which should eventually be up.